ci-operator: set TTL annotations before everything else

petr-muller opened the pull request:
ci-operator was setting up SSAR and pull secrets between the namespace
creation and setting up the annotations, so if one of them failed, the
annotations were not set and we leak the namespace.

alvaroaleman commented on cmd/ci-operator/main.go:
```diff
@@ -943,6 +892,57 @@ func (o *options) initializeNamespace() error {
 		}
 	}
 
+	ssarStart := time.Now()
```
Can we not simply set the annotation on creation or does the projects api not allow that?

petr-muller replied:
Possibly, haven't tried. It's not entirely braindead because the annotations are technically optional and you may still need to update afterwards because you are reusing an existing project but it would be a safest thing to do.

I'm not too keen on doing it myself though - reordering stuff seemed easy enough to just do it, but I'd really like to avoid another rabbit hole :)

alvaroaleman replied:
I guess the point is, this might just fail because the SSAR succeeded and is generally still prone to racing

alvaroaleman replied:
The ProjectRequest takes an ObjectMeta, lets just add the annotation there and then do the update after SSAR. We still have to do the update in case we didn't create the project

petr-muller replied:
OK, will fixup the PR, stay tuned.

smarterclayton replied:
You can't set annotations on project (historical limitation).  So if you send them they'll be ignored I think (but if you see in your test they don't let me know)

stevekuznetsov replied:
Let's add a comment, when I wrote this it was common knowledge

stevekuznetsov commented:
/hold

wait

We can't do *anything* before the SSAR check goes through, as that's when we actually have RBAC in the namespace to do anything in the first place. Please add the annotation logic after this check.

stevekuznetsov reviewed (approved):
/hold cancel

petr-muller commented:
/hold cancel

Addressed. Moved the annotation setup after SSAR but before pull secret setup. Added a comment explaining the quirks involved.

I confirmed that adding annotations/labels does not work in the initial `Create` API call.
